pkg/cmd/operator/mustgather.go

```diff
@@ -0,0 +1,290 @@
+}
+
+type resourceSpec struct {
+	collect.ResourceInfo
+	Namespace, Name string
+}
+
+var mustGatherSpecs = []struct {
+	schema.GroupResource
+	Namespace, Name string
+}{
+	{
+		GroupResource: schema.GroupResource{
+			Resource: "scyllaclusters",
+			Group:    "scylla.scylladb.com",
+		},
+		Namespace: corev1.NamespaceAll,
+		Name:      "",
+	},
+	{
+		GroupResource: schema.GroupResource{
+			Resource: "scyllaoperatorconfigs",
+			Group:    "scylla.scylladb.com",
+		},
+		Namespace: corev1.NamespaceAll,
+		Name:      "",
+	},
+	{
+		GroupResource: schema.GroupResource{
+			Resource: "nodeconfigs",
+			Group:    "scylla.scylladb.com",
+		},
+		Namespace: corev1.NamespaceAll,
+		Name:      "",
+	},
+	{
+		GroupResource: schema.GroupResource{
+			Resource: "namespaces",
+			Group:    "",
+		},
+		Namespace: corev1.NamespaceAll,
+		Name:      "scylla-operator",
+	},
+	{
+		GroupResource: schema.GroupResource{
+			Resource: "namespaces",
+			Group:    "",
+		},
+		Namespace: corev1.NamespaceAll,
+		Name:      "scylla-manager",
+	},
+	{
+		GroupResource: schema.GroupResource{
+			Resource: "validatingwebhookconfigurations",
+			Group:    "admissionregistration.k8s.io",
+		},
+		Namespace: corev1.NamespaceAll,
+		Name:      "",
+	},
+	{
+		GroupResource: schema.GroupResource{
+			Resource: "mutatingwebhookconfigurations",
+			Group:    "admissionregistration.k8s.io",
+		},
+		Namespace: corev1.NamespaceAll,
+		Name:      "",
+	},
+}
+
+func (o *MustGatherOptions) run(ctx context.Context) error {
+	startTime := time.Now()
+	klog.InfoS("Gathering artifacts", "TargetDir", o.DestDir)
+	defer func() {
+		klog.InfoS("Finished gathering artifacts", "Duration", time.Since(startTime))
+	}()
+
+	collector := collect.NewCollector(
+		o.DestDir,
+		o.GetPrinters(),
+		o.discoveryClient,
+		o.kubeClient.CoreV1(),
+		o.dynamicClient,
+		true,
+		o.KeepGoing,
+	)
+
+	var resourceSpecs []resourceSpec
+	if o.AllResources {
+		allPreferredListableResources, err := collector.DiscoverResources(ctx, discovery.SupportsAllVerbs{
+			Verbs: []string{"list"},
+		}.Match)
+		if err != nil {
+			return err
+		}
+
+		// Filter out native resources that share storage across groups.
+		preferredListableResources, err := collect.ReplaceIsometricResourceInfosIfPresent(allPreferredListableResources)
+		if err != nil {
+			return fmt.Errorf("can't replace isometric resourceInfos: %w", err)
+		}
+
+		resourceSpecs = make([]resourceSpec, 0, len(preferredListableResources))
+		for _, pr := range preferredListableResources {
+			resourceSpecs = append(resourceSpecs, resourceSpec{
+				ResourceInfo: collect.ResourceInfo{
+					Resource: pr.Resource,
+					Scope:    pr.Scope,
+				},
+				Namespace: corev1.NamespaceAll,
+				Name:      "",
+			})
+		}
+	} else {
+		preferredResources, err := collector.DiscoverResources(ctx, func(groupVersion string, r *metav1.APIResource) bool {
+			return true
+		})
+		if err != nil {
+			return err
+		}
+
```

Comment:
I don't think that would be fitting here, at least not at this point. In this case the set is not coming from discovery but desired specs. We shouldn't ask for duplicate resources in our specs. Maybe if we need some of them to be optional at some point in the future we could extend the search across groups and then it would make more sense.